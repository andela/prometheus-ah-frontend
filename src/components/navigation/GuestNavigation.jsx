import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import ModalForm from '../modals/Modal';


/**
 * @description class for userNavigation header
 *
 * @class GuestNavigation
 *
 * @extends {Component}
 */
class GuestNavigation extends Component {
  /**
   * @description Render the JSX template
   *
   * @returns {JSX} render JSX template
   *
   * @memberof Header
   */
  render() {
    const { showLoginModal, showSignupModal, modal } = this.props;
    return (
      <nav className="navbar-nav">
        <li className="nav-item">
          <a className="nav-link" href="/">Home</a>
        </li>
        <li className="nav-item">
          <Link
            to="#"
            className="nav-link"
            onClick={showLoginModal}
          >
            Login
          </Link>
          <ModalForm modal={modal} />
        </li>
        <li className="nav-item" id="signup">
          <Link
            to="#"
            className="nav-link"
            onClick={showSignupModal}
          >
            Sign Up
          </Link>
          <ModalForm modal={modal} />
        </li>
      </nav>
    );
  }
}

GuestNavigation.propTypes = {
  showLoginModal: PropTypes.func,
  showSignupModal: PropTypes.func,
  modal: PropTypes.shape({})
};

export default GuestNavigation;
