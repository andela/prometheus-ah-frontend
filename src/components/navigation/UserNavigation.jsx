import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import logoutAction from '../../actions/authentication/logoutAction';

/**
 * @description class for userNavigation header
 *
 * @class UserNavigation
 *
 * @extends {Component}
 */
export class UserNavigation extends Component {
  /**
   * @description handle user log out
   *
   * @param {Object} event logout event object
   *
   * @memberof Header
   *
   * @returns {undefined} calls logoutProps
   */
  onLogout = (event) => {
    const { logoutUser } = this.props;
    event.preventDefault();
    logoutUser();
  }

  /**
   * @description Render the JSX template
   *
   * @returns {JSX} render JSX template
   *
   * @memberof Header
   */
  render() {
    const { authUsername } = this.props;

    return (
      <ul className="navbar-nav">
        <li className="nav-item">
          <a className="nav-link" href="/">Home</a>
        </li>
        <li className="nav-item">
          <a className="nav-link" href="/">
            { authUsername }
          </a>
        </li>
        <li className="nav-item" id="login">
          <Link
            to="#"
            id="logout"
            className="nav-link"
            onClick={this.onLogout}
          >
            Logout
          </Link>
        </li>
      </ul>
    );
  }
}


UserNavigation.propTypes = {
  logoutUser: PropTypes.func,
  authUsername: PropTypes.string
};

const mapDispatchToProps = dispatch => ({
  logoutUser: () => dispatch(logoutAction()),
});

const mapStateToProps = state => ({
  authUsername: state.auth.user.username
});

export default connect(mapStateToProps, mapDispatchToProps)(UserNavigation);
