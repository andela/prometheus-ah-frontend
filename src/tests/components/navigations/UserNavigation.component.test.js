import React from 'react';
import { shallow } from 'enzyme';
import { UserNavigation } from '../../../components/navigation/UserNavigation';
import mockData from '../../__mocks__/mockData';


let wrapper;

describe('UserNavigation component', () => {
  beforeEach(() => {
    const { user } = mockData;
    const props = {
      logoutUser: jest.fn(),
      user
    };
    wrapper = shallow(<UserNavigation {...props} />);
  });

  test('should render a snapshot of the userNavigation component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should renders a snapshot of the userNavigation component', () => {
    // const logoutBtn = wrapper.find('#logout');
    wrapper.find('#logout').simulate('click', {
      preventDefault: jest.fn(),
    });

    wrapper.setState({
      user: {}
    });

    expect(wrapper.state('user')).toEqual({});
  });
});
