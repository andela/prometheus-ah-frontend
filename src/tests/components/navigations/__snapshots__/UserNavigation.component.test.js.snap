// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestNavigation componete should renders a snapshot of the guestNavigation component 1`] = `
<ul
  className="navbar-nav"
>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/"
    >
      Home
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/"
    />
  </li>
  <li
    className="nav-item"
    id="login"
  >
    <Link
      className="nav-link"
      id="logout"
      onClick={[Function]}
      replace={false}
      to="#"
    >
      Logout
    </Link>
  </li>
</ul>
`;
